src/libos/src/process/do_spawn/init_stack.rs

```diff
@@ -53,11 +54,13 @@ pub fn do_init(
     init_area_size: usize,
     argv: &[CString],
     envp: &[CString],
-    auxtbl: &AuxVec,
+    auxtbl: &mut AuxVec,
 ) -> Result<usize> {
     let stack_buf = unsafe { StackBuf::new(stack_top, init_area_size)? };
     let envp_cloned = clone_cstrings_on_stack(&stack_buf, envp)?;
     let argv_cloned = clone_cstrings_on_stack(&stack_buf, argv)?;
+    let rand_val_ptr = generate_random_on_stack(&stack_buf)?;
+    auxtbl.set(AuxKey::AT_RANDOM, rand_val_ptr as *const () as u64);
```

Comment:
This is because the AuxKey::AT_RANDOM in aux table is a pointer that points to a value in stack. The pointer must be initialized on the stack before setting the aux table.

To make the aux_tbl immutable, maybe I can change the stack from one step to two steps: 
init_stack_phase1() --> aux_table::init(random_value_ptr_on_stack) --> init_stack_phase2()

Is it make sense?

Reply:
I see. Then I suggest keeping the current code.